src/main/java/teammates/ui/controller/StudentProfilePictureAction.java

```diff
@@ -42,8 +43,16 @@ private ActionResult handleRequestWithBlobKey() {
 
     private ActionResult handleRequestWithEmailAndCourse()
             throws EntityDoesNotExistException {
-        String email = getStudentEmailFromRequest();
-        String courseId = getCourseIdFromRequest();
+        String email;
+        String courseId;
+        try {
+            email = getStudentEmailFromRequest();
+            courseId = getCourseIdFromRequest();
+        } catch (InvalidParametersException e) {
+            // potential PMD bug? No stack trace info is lost here
+            throw new EntityDoesNotExistException(e.getMessage()); //NOPMD
```

Comment:
Is this a `EntityDoesNotExistException`? should there be a log message instead/as well?